src/Accelerators/NNPA/Conversion/ONNXToZHigh/RewriteONNXForZHigh.cpp

```diff
@@ -216,6 +217,94 @@ bool CanExpandPowOpToMul(ONNXPowOp op) {
+  ONNXConvOpAdaptor operandAdaptor = ONNXConvOpAdaptor(op);
+  ONNXConvOpShapeHelper shapeHelper(&op);
+  assert(succeeded(shapeHelper.computeShape(operandAdaptor)));
+  return (shapeHelper.pads.size() == 4) &&
+         (llvm::all_of(
+             shapeHelper.pads, [](IndexExpr val) { return val.isLiteral(); }));
+}
+
+// Create an ArrayAttr of IntergerAttr(s) of zero values.
+// This function is used for padding attribute in Conv.
+ArrayAttr getPadsForNNPAConv(PatternRewriter &rewriter, Value ret) {
+  ONNXConvOp op = dyn_cast<ONNXConvOp>(ret.getDefiningOp());
+  ONNXConvOpAdaptor operandAdaptor = ONNXConvOpAdaptor(op);
+  ONNXConvOpShapeHelper shapeHelper(&op);
+  assert(succeeded(shapeHelper.computeShape(operandAdaptor)));
+  SmallVector<int64_t, 4> vals = {0, 0, 0, 0};
+  for (size_t i = 0; i < shapeHelper.pads.size(); i++) {
+    vals[i] = shapeHelper.pads[i].getLiteral();
+  }
+  return rewriter.getI64ArrayAttr(vals);
+}
+
+// Pad a ArrayAttr with zeros.
+//
+// pads = [B1, B2, ... Bk, E1, E2, ..., Ek]
+//
+// becomes:
+//
+// pads = [0,... 0, B1, B2, ... Bk, 0,... 0, E1, E2, ..., Ek]
+//         |_____|                  |_____|
+//                 nZeros                    nZeros
+//
+// This function is used for padding attribute in Conv.
+DenseElementsAttr insertZerosForNonPaddedDims(
+    PatternRewriter &rewriter, ArrayAttr origAttrs, int extensionLength) {
+  int nDims = (int)origAttrs.getValue().size() / 2;
+  int nElements = (nDims + extensionLength) * 2;
+  SmallVector<int64_t, 4> pads(nElements, 0);
+  for (int i = 0; i < nDims; ++i) {
+    int64_t beginPad = origAttrs.getValue()[i].cast<IntegerAttr>().getInt();
+    int64_t endPad =
+        origAttrs.getValue()[nDims + i].cast<IntegerAttr>().getInt();
+    pads[i + extensionLength] = beginPad;
+    pads[nDims + extensionLength + i + extensionLength] = endPad;
+  }
+  return rewriter.getI64TensorAttr(llvm::makeArrayRef(pads));
+}
+
+DenseElementsAttr createDenseFloatAttrOfValue(
+    PatternRewriter &rewriter, Value origValue, float constantValue) {
+  Type elementType = origValue.getType().cast<TensorType>().getElementType();
+  SmallVector<float, 1> wrapper(1, 0);
+  wrapper[0] = constantValue;
+  return DenseElementsAttr::get(
+      RankedTensorType::get({}, elementType), llvm::makeArrayRef(wrapper));
+}
+
+// Create an ArrayAttr of IntergerAttr(s) of zero values.
+// This function is used for padding attribute in Conv.
+ArrayAttr createArrayAttrOfZeros(
+    PatternRewriter &rewriter, ArrayAttr origAttrs) {
+  int nElements = origAttrs.getValue().size();
+  SmallVector<int64_t, 4> vals(nElements, 0);
+  return rewriter.getI64ArrayAttr(vals);
+}
+
+// Create Type for Padded input
+Type CreatePaddedXType(Value x, ArrayAttr pads) {
+  RankedTensorType inputType = x.getType().cast<RankedTensorType>();
+  ArrayRef<int64_t> inputShape = inputType.getShape();
+  Type elementType = inputType.getElementType();
+  SmallVector<int64_t, 4> paddingShape = {0, 0, 0, 0};
```

Comment:
Use `SmallVector<int64_t, 4> paddingShape(4, 0)` for brevity.

Reply:
Fixed. Thank you!